macros/sql/get_column_values.sql

```diff
@@ -36,16 +37,26 @@ Returns:
 
         {%- else -%}
 
-            select
-                {{ column }} as value
-
-            from {{ target_relation }}
-            group by 1
-            order by count(*) desc
-
-            {% if max_records is not none %}
-            limit {{ max_records }}
-            {% endif %}
+            with sorted_column_values as (
+
+                select
+                    {{ column }} as value,
+                    {%- if sort_column is none %}
+                    count(*) as sort_column
+                    {% else %}
+                    {# We take the max sort value for each value to make sure 
+                        there are no duplicate rows for each value #}
+                    max({{ sort_column }}) as sort_column
```

Comment:
What do you think of instead doing something like:
```
{% set order_by = kwargs.get('order_by', 'count(*)')
```
Then people could also order alphabetically by providing:
```
{% set states = dbt_utils.get_column_values(table=ref('users'), column='state', order_by='state') %}

```

And by recency like so:
```
{% set states = dbt_utils.get_column_values(table=ref('users'), column='state', order_by='min(created_at)') %}
```

Reply:
I like it! I'll push a commit in a sec. Thanks!